src/components/KonnectorSuccess.jsx

```diff
@@ -77,35 +69,95 @@ export class KonnectorSuccess extends Component {
           >
             {hasLinks && (
               <SuccessLinks>
-                {displayDriveUrl && <DriveLink folderId={trigger.message.folder_to_save} />}
-                {displayBanksUrl && <BanksLink banksUrl={banksUrl} />}
+                {relatedApps.map((x, i) =>
+                  x.successLink(this.props, this.context, i)
+                )}
               </SuccessLinks>
             )}
           </DescriptionContent>
 
           <SuccessFooter>
-            <Button
-              label={successButtonLabel || t('account.success.button')}
-              onClick={event => {
-                event.preventDefault()
-                onDone(account)
-              }}
-            />
+            {relatedApps.length > 0
+              ? relatedApps[0].footerLink(this.props, this.context)
+              : null}
           </SuccessFooter>
         </div>
       )
     )
   }
 }
 
+KonnectorSuccess.apps = {
+  drive: {
+    priority: 1,
+    // eslint-disable-next-line react/display-name
+    predicate: props => {
+      const trigger = props.trigger
+      const res = has(trigger, 'message.folder_to_save')
+      return res
+    },
+    // eslint-disable-next-line react/display-name
+    successLink: (props, context, i) => {
+      return (
+        <DriveLink key={i} folderId={props.trigger.message.folder_to_save} />
+      )
+    },
+    // eslint-disable-next-line react/display-name
+    footerLink: props => {
+      const { t, onDone, account, successButtonLabel } = props
+      return (
+        <Button
+          label={successButtonLabel || t('account.success.button')}
+          onClick={event => {
+            event.preventDefault()
+            onDone(account)
+          }}
+        />
+      )
+    }
+  },
+
+  banks: {
+    priority: 0,
+    // eslint-disable-next-line react/display-name
+    predicate: props => {
+      const connector = props.connector
+      return (
+        Array.isArray(connector.data_types) &&
+        connector.data_types.includes('bankAccounts')
+      )
+    },
+    // eslint-disable-next-line react/display-name
+    successLink: (props, context, i) => {
+      return <BanksLink key={i} banksUrl={props.store.banksUrl} />
+    },
+    footerLink: () => null
+  }
+}
+
 KonnectorSuccess.propTypes = {
   account: PropTypes.object.isRequired,
-  error: PropTypes.string.isRequired,
+  error: PropTypes.string,
   title: PropTypes.string.isRequired,
   messages: PropTypes.arrayOf(PropTypes.string),
   onDone: PropTypes.func.isRequired,
   successButtonLabel: PropTypes.string.isRequired,
   trigger: PropTypes.object.isRequired
 }
 
-export default translate()(KonnectorSuccess)
+export {
+  SuccessImage,
+  SuccessLinks,
+  BanksLink,
+  DriveLink
+}
+
+const provideStoreAsProp = connect(
+  null,
+  (dispatch, store) => ({ store })
+)
+
+export default compose(
+  translate(),
+  provideStoreAsProp
+)(KonnectorSuccess)
```

Comment:
`translate()(provideStoreAsProp(KonnectorSuccess))`
Using compose/flowRight here is less readable than the line above and doesn't add any value

Reply:
The compose pattern is used throughout our apps so I'd guess it would be pretty readable. To have each HOC on its own line enables to have clear diffs also.

Reply:
I changed it in https://github.com/cozy/cozy-home/pull/1204/commits/e533ddaccb48a9651dbd43d2df03902afcbb7fb0 but I feel like it's nitpicking and a non issue.